pkg/controller/messaginginfra/cert/controller.go

```diff
@@ -180,22 +345,39 @@ func (c *CertController) applyCertSecret(secret *corev1.Secret, caSecret *corev1
 	logger.Info("Checking cert", "secret", secret.Name, "expiryDate", expiryDate.Format(time.UnixDate), "expectedCaDigest", expectedCaDigest, "actualCaDigest", actualCaDigest)
 
 	now := c.clock.Now()
+	var certInfo *CertInfo
 
 	// Create the initial certificate if this is a new secret to be created
 	if !hasCert {
 		logger.Info("Creating component certificate", "secret", secret.Name)
 		expiryDate := now.Add(c.certExpirationTime)
-		key, cert, keystore, truststore, err := generateCert(nil, nil, caSecret.Data["tls.key"], caSecret.Data["tls.crt"], expiryDate, commonName, dnsNames)
+		key, cert, keystore, truststore, err := generateCert(now, nil, nil, caSecret.Data["tls.key"], caSecret.Data["tls.crt"], expiryDate, commonName, dnsNames)
```

Comment:
I wonder if we should generate the cert with a valid from 1h in the past, to help guard against clock skew causing spurious connection failures.